src/Control/Monad/Free/TH.hs

```diff
@@ -3,7 +3,11 @@
 {-# OPTIONS_GHC -Wno-overlapping-patterns #-}
 #endif
 #include "free-common.h"
-
+#if __GLASGOW_HASKELL__ >= 704 && MIN_VERSION_template_haskell(2,12,0)
```

Comment:
`free` doesn't support any versions of GHC before 7.4, so the `__GLASGOW_HASKELL__ >= 704` part is redundant.

Reply:
done.